src/query/more_like_this/query.rs

```diff
@@ -22,26 +24,61 @@ use crate::DocAddress;
 ///     .with_stop_words(vec!["for".to_string()])
 ///     .with_document(DocAddress::new(2, 1));
 /// ```
-#[derive(Debug, Clone)]
-pub struct MoreLikeThisQuery {
+pub struct MoreLikeThisQuery<D: DocumentAccess = Document> {
```

Comment:
Not sure if this worth the added complexity. The trait is mainly for performance reasons during indexing

Reply:
I agree, after using it a bit the convenience when indexing is great but getting a document back from a searcher kind of sucks :/ MoreLikeThisQuery also gets caught up in the generic mess because it takes in a value which now can also be generic is a user is fetching other docs.

I'm considering if it's best to box this to hide the generic as it really did make it pretty annoying to work with.

Reply:
The serialization should already be the same, which is why you can serialize as one type with `DocumentAccess` but deserialize as `Document` still.

The problem I see though is that if we only support serialization with a custom type, people will likely be confused as to why you can't then retrieve a custom type back from the search.

As I write this I think because of the first point I mentioned we can simply remove the generic from `MoreLikeThis` and just deserialize it as a Document (Which may be what you're saying already in which case I apologise for the roundabout conversion)

Reply:
Ended up fixing this and just using the default Document types internally.